core/templates/components/checkpoint-celebration-modal/checkpoint-celebration-modal.component.ts

```diff
@@ -0,0 +1,312 @@
+    private explorationPlayerStateService: ExplorationPlayerStateService
+  ) {}
+
+  ngOnInit(): void {
+    this.explorationId = this.contextService.getExplorationId();
+    this.oppiaAvatarImageUrl = (
+      this.urlInterpolationService
+        .getStaticImageUrl('/avatar/oppia_avatar_100px.svg'));
+    this.readOnlyExplorationBackendApiService.fetchExplorationAsync(
+      this.explorationId, null).then((response) => {
+      this.exploration = response.exploration;
+      this.hasViewedLessonInfoOnce = response.has_viewed_lesson_info_modal_once;
+      this.mostRecentlyReachedCheckpointStateName = (
+        response.most_recently_reached_checkpoint_state_name);
+
+      this.orderedCheckpointList = this.checkpointCelebrationUtilityService
+        .getStateListForCheckpointMessages(
+          this.exploration.states, this.exploration.init_state_name);
+      this.totalNumberOfCheckpoints = this.orderedCheckpointList.length;
+      this.checkpointStatusArrayPlaceholder = new Array(
+        this.totalNumberOfCheckpoints);
+      this.checkpointStatusArray = new Array(this.totalNumberOfCheckpoints);
+      this.checkpointNodeFadeInDelays = new Array(
+        this.totalNumberOfCheckpoints);
+      this.setFadeInDelaysForCheckpointNodes();
+      this.currentStateName = this.exploration.init_state_name;
+      this.subscribeToCardChangeEmitter();
+    });
+    this.shouldDisplayFullScaleMessage = (
+      this.windowDimensionsService.getWidth() > 1370);
+    this.subscribeToWindowResizeEmitter();
+  }
+
+  ngOnDestroy(): void {
+    this.directiveSubscriptions.unsubscribe();
+  }
+
+  subscribeToCardChangeEmitter(): void {
+    this.directiveSubscriptions.add(
+      this.playerPositionService.onNewCardOpened.subscribe(
+        (nextStateCard: StateCard) => {
+          this.checkpointCelebrationUtilityService.setIsOnCheckpointedState(
+            false);
+          if (this.miniMessageTooltipIsShown) {
+            this.dismissMiniMessage();
+            setTimeout(() => {
+              this.checkIfCheckpointMessageIsToBeTriggered(
+                nextStateCard.getStateName());
+            }, MESSAGE_MODAL_APPROX_TRIGGER_AND_DISMISSAL_DURATION_MS);
+          } else if (this.messageModalIsShown) {
+            this.dismissMessage();
+            setTimeout(() => {
+              this.checkIfCheckpointMessageIsToBeTriggered(
+                nextStateCard.getStateName());
+            }, MESSAGE_MODAL_APPROX_TRIGGER_AND_DISMISSAL_DURATION_MS);
+          } else {
+            this.checkIfCheckpointMessageIsToBeTriggered(
+              nextStateCard.getStateName());
+          }
+        }
+      )
+    );
+  }
+
+  subscribeToWindowResizeEmitter(): void {
+    this.directiveSubscriptions.add(
+      this.windowDimensionsService.getResizeEvent().subscribe(
+        () => {
+          this.shouldDisplayFullScaleMessage = (
+            this.windowDimensionsService.getWidth() > 1370);
```

Comment:
Should we use a constant for the number?

Reply:
Done